design/fips-compliance.md

```diff
@@ -0,0 +1,256 @@
+# FIPS compliance for conjur-authn-k8s-client
+
+## Background
+
+The National Institute of Standards and Technology (NIST) issued the [FIPS
+ 140-2](https://csrc.nist.gov/publications/detail/fips/140/2/final) 
+Publication Series to coordinate the requirements and standards for cryptography 
+modules that include both hardware and software components.
+
+By making the conjur-authn-k8s-client FIPS compliant, our customer will gain a 
+cryptographic standard to protect their unclassified, but sensitive data.
+
+## Useful Links
+
+| **Name**                        | **Link**                                            |
+|---------------------------------|-----------------------------------------------------|
+| Epic - FIPS compliance for conjur-authn-k8s-client | https://github.com/cyberark/conjur-authn-k8s-client/issues/91 |
+| FIPS compliant crypto in golang | https://kupczynski.info/2019/12/15/fips-golang.html |
+
+## Solution
+
+We looked into 3 options for making our product FIPS compliant:
+
+1. Use cgo to call out to an existing certified library
+   
+    - This alternative disqualified due to its complexity and consuming time
+     and effort as we will have to replace any usage of crypto.
+     
+1. Use [RHEL go toolchain](https://developers.redhat.com/blog/2019/06/24/go-and-fips-140-2-on-red-hat-enterprise-linux/).
+   - RHEL takes the ownership on the encryption, and bridges between the
+    Golang encryption and OpenSSL.
+   - What needs to be done is build the Golang project on subscribed RHEL
+    machine/UBI container with the relevant go-toolset installed on it. Then, 
+    we can take the compiled binary and copy it into the containers as usual.
+   - A big advantage of this approach is that we are aligning with Red Hat
+    which is one of the big players in the field of certifications, enterprise, 
+    and federal market.
+   - The pipeline should use and subscribe RHEL / UBI on-the-fly.
+   
+1. Use BoringSSL based crypto
+   Use [Google’s fork of Golang](https://github.com/golang/go/blob/dev.boringcrypto.go1.12/misc/boring/README.md) 
+   that uses dev.boringcrypto as its crypto engine. dev.boringcrypto wraps BoringSSL which is FIPS compliant.
```

Comment:
> which is FIPS compliant

has this been verified? Do we know that using this library is sufficient for us to be FIPS certified?

If it has been verified, can we please link to the source on that here?

Reply:
it is mentioned in the `Security` section. I'll link to that.